### DISCUSSION TITLE
RuntimeBinderException: List doesn't contain definition for Select (Map-Reduce)

### DISCUSSION THREAD

**Discussion by jackabouaad-medel** [ravendb/ravendb#15146] [Q&A]
Hi,
I am getting a RuntimeBinderException _'System.Collections.Generic.List<string>' does not contain a definition for 'Select'_ error when trying to execute the Map-Reduce shown below:
```
Map = words =>
            from token in words
            let split = token.transcription.Split().ToList()
            from pair in split.Select((value, index) => new { value, index })
            select new Result
            {
                Key = string.Join(' ', split.Select((v, i) => i == pair.index ? "_" : v)),
                Pairs = new List<string>() { token.word }
            };
```
Note that I referenced the System.Linq assembly as follows:
```
AdditionalAssemblies = new() {AdditionalAssembly.FromNuGet("System.Linq", "4.3.0", usings: new() {"System.Linq"})};
```
Am I missing something, or is it impossible to use List within a Map-Reduce?

**Comment by maciejaszyk** [comment]
Hi,
Why do you need to import LINQ from NuGet?

BTW:
I see you're using `string.Join(char,....)`. 
There is an issue with that (please take a look here: https://github.com/ravendb/ravendb/discussions/14032#discussioncomment-2530274 )

Can you try with `" "` instead of `' '`?

**Reply by jackabouaad-medel** [reply]
Because I googled the issue and all the answers point to a mis-reference of System.Linq, even though I am writing the whole thing in LINQ.

I tried substituting the single quotes with a double quotes, but the issue is still the same.

**Comment by ayende** [comment]
You should use an array, instead:  `Pairs = new []{ token.word }`

Types don't really mean what you think they do server side.

**Reply by jackabouaad-medel** [reply]
My issue is with the split variable, the Split method is returning an Array<string>, but for some reason I cannot cast the Array<string> to List<string>. Therefore I cannot do a mapping (Select) on the split variable itself.

**Reply by ppekrol** [reply]
I've created following index in database with our Sample Data.  I'm creating that list via ctor. I think it should solve your issue:

```
from o in docs.Orders
let split = new List<string>(o.Company.Split('-'))
from pair in split.Select((value, index) => new { value, index })
select new 
{
    Key = string.Join(' ', split.Select((v, i) => i == pair.index ? "_" : v)),
    Pairs = new List<string>() { o.Company }
}
```
